src/FakeItEasy/Creation/DummyValueResolver.cs

```diff
@@ -66,12 +65,12 @@ private bool TryResolveDummyValueWithAllAvailableStrategies(DummyCreationSession
             {
                 if (strategy.TryCreateDummyValue(session, typeOfDummy, out result))
                 {
-                    this.strategyCache.Add(typeOfDummy, strategy);
+                    this.strategyCache.TryAdd(typeOfDummy, strategy);

```

Comment:
What if it fails? Just ignore it, and we'll have to resolve the correct strategy again next time?